core-backend/wasmtime/src/env.rs

```diff
@@ -137,28 +171,32 @@ impl<E: Ext + IntoExtInfo> Environment<E> for WasmtimeEnvironment<E> {
         funcs.insert("gr_wait", funcs::wait(&mut store));
         funcs.insert("gr_wake", funcs::wake(&mut store, memory));
 
-        let module = Module::new(&engine, binary).map_err(|e| BackendError {
-            reason: "Unable to create module",
-            description: Some(e.to_string().into()),
-            gas_amount: later_ext.unset().into_gas_amount(),
-        })?;
-
-        let mut imports = module
-            .imports()
-            .map(|import| {
-                if import.module() != "env" {
-                    Err(BackendError {
-                        reason: "Non-env imports are not supported",
-                        description: import
-                            .name()
-                            .map(|v| format!("Function {:?} is not env", v).into()),
-                        gas_amount: later_ext.unset().into_gas_amount(),
-                    })
-                } else {
-                    Ok((import.name(), Option::<Extern>::None))
-                }
-            })
-            .collect::<Result<Vec<_>, BackendError>>()?;
+        let module = match Module::new(&engine, binary) {
+            Ok(module) => module,
+            Err(e) => {
+                return Err(BackendError {
+                    reason: "Unable to create module",
+                    description: Some(e.to_string().into()),
+                    gas_amount: get_current_gas_state(later_ext)
+                        .expect("existing clone is not taken"),
+                })
+            }
+        };
+
+        let mut imports = Vec::with_capacity(module.imports().len());
```

Comment:
What `with_capacity` is for here?

Reply:
Sorry, could you clarify the purpose of the question, so I can answer it precisely

Reply:
I can't get what you want to ask.

Reply:
Sorry, could you clarify the purpose of the question, so I can answer it precisely

Reply:
https://github.com/gear-tech/gear/pull/836#discussion_r849439289

Reply:
I got answer in previous conversation, so nevermind